cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+				logrus.Warnf("Attempted to remove Grafana service after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Grafana service with name: %v", grafanaService.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Grafana service")
+	}
+
+	shouldRemoveLokiDeployment = false
+	shouldRemoveGrafanaConfigMap = false
+	shouldRemoveGrafanaDeployment = false
+	shouldRemoveGrafanaService = false
+	shouldRemoveNamespace = false
+	shouldRemoveLokiService = false
+	return lokiHost, nil
+}
+
+func checkGrafanaAndLokiDeploymentExistence(ctx context.Context, k8sManager *kubernetes_manager.KubernetesManager) (bool, string) {
+	existsLoki := false
+	existsGrafana := false
+	var lokiHost string
+	lokiDeployment, err := k8sManager.GetDeployment(ctx, graflokiNamespace, lokiDeploymentName)
+	if err == nil && lokiDeployment != nil {
+		existsLoki = true
+		lokiHost = getLokiHost(lokiServiceName, graflokiNamespace, lokiNodePort)
+	}
+
+	grafanaDeployment, err := k8sManager.GetDeployment(ctx, graflokiNamespace, grafanaDeploymentName)
+	if err == nil && grafanaDeployment != nil {
+		existsGrafana = true
+	}
+
+	return existsLoki && existsGrafana, lokiHost
+}
+
+func StopGrafLokiInKubernetes(ctx context.Context) error {
+	k8sManager, err := getKubernetesManager()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting Kubernetes Manager.")
+	}
+	ns, err := k8sManager.GetNamespace(ctx, graflokiNamespace)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting graflokiNamespace '%v'.", graflokiNamespace)
+	}
+	err = k8sManager.RemoveNamespace(ctx, ns)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred removing graflokiNamespace '%v'.", graflokiNamespace)
+	}
+	err = waitForNamespaceRemoval(ctx, graflokiNamespace, k8sManager)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while waiting for graflokiNamespace '%v' removal.", graflokiNamespace)
+	}
+	return nil
+}
+
+func getKubernetesManager() (*kubernetes_manager.KubernetesManager, error) {
+	kubernetesConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(), nil,
+	).ClientConfig()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating Kubernetes configuration")
+	}
+	clientSet, err := kubernetes.NewForConfig(kubernetesConfig)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Expected to be able to create Kubernetes client set using Kubernetes config '%+v', instead a non nil error was returned", kubernetesConfig)
+	}
+	k8sManager := kubernetes_manager.NewKubernetesManager(clientSet, kubernetesConfig, "")
+	return k8sManager, nil
+}
+
+func waitForNamespaceRemoval(
+	ctx context.Context,
+	namespace string,
+	kubernetesManager *kubernetes_manager.KubernetesManager) error {
+	var (
```

Comment:
can this be `const`? remove cyclomatic complexity?

Reply:
i like putting them here bc its closer to the context they're used in